### DISCUSSION TITLE
Tips on getting rclone working with Google Drive?

### DISCUSSION THREAD

**Discussion by DSew20** [outdoorbits/little-backup-box#300] [General]
I've set up a RPi 4B based LBB saving to SSD. All seems to be working well.

I would also like to backup to the Cloud. I've followed the instructions here:

https://github.com/outdoorbits/little-backup-box/wiki/04b.-Cloud-services

Has anybody got any further tips / guides on how to get this working?

**Comment by outdoorbits** [comment]
It is working here. What's the point you are now?

**Reply by DSew20** [reply]
I downloaded rclone to windows. Created a new remote, names, followed the various options (default for most), was taken to Google Drive, logged in and rclone updated rclone.conf (saved in C:\Users\[user]\AppData\Roaming\rclone)

I took the config code from the above and added to the rclone.conf file in LBB root. Then rebooted LBB. But, nothing seems to happen!

**Comment by outdoorbits** [comment]
Hopefully you are already there. There is no reboot required. On the web 
UI, have a look at the backup page. Are there new (blue) options for 
source and target drives? Just use them. You also can configure them as 
default backup drives. Is it working?

**Reply by DSew20** [reply]
Thank you.

I have tested the rclone config in the cmd prompt on Windows and can see the files - all seems ok.

I have copied the rclone.conf content to lbb and can see the blue text. But, when I try to mount or backup to the rclone drive I get the below in the log:

```
2024-06-18 19:21:00
MOUNTED?: '/media/cloud_target' = False

2024-06-18 19:21:00
umount /media/cloud_target: Not mounted.

2024-06-18 19:21:00
MOUNTED?: '/media/cloud_target' = False
```

**Comment by outdoorbits** [comment]
Strange. But if you use this configuration file on your windows machine, 
you still can access your cloud drive?

Maybe you get more information about the error, when you go to lbb - 
Settings - CONFIGURE CLOUD-SERVICES and open rclone GUI by following the 
link. Here you can try to connect using Explorer. Hopefully you get an 
error message.

What does your GoogleDrive section look like? This is mine:

[GoogleDrive]
type = drive
scope = drive
root_folder_id =
token = 
{"access_token":"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX","token_type":"Bearer","refresh_token":"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX","expiry":"XXXX-XX-18T21:49:38.083882565+02:00"}
team_drive =

**Reply by DSew20** [reply]
I added the root_folder_id = line as this was missing from my config - setup again and seems ok now.

Thank you for sharing the config.

**Comment by outdoorbits** [comment]
Maybe this was a problem about windows format new line?

**Reply by DSew20** [reply]
A related question - any success getting LBB to sync to Google Photos?

I ask as Photos would allow viewing RAW files whereas Drive does not. I have setup an rclone for Photos. LBB seems to connect, but Photos are not copied.

Just wonder if others had any success? Or, is there a trick needed re LBB Settings / Configure Cloud Services / Google Photos / Directories as Photos uses "Albums" (which may be treated differently to Directories)?

**Comment by outdoorbits** [comment]
I just tested it more or less successfully. In rclone config I set read_size to true (andvanced options) everything else to default. I'm not sure if this is a useful setup. In lbb Settings - Configure Cloud-services - Base folder on cloud servers set "upload".
Upload works but GooglePhotos immediately moves the uploaded files. I think that's why you always get an error.  Because you can't easily get the original resolution of the image from GooglePhotos, this is maybe not a very useful destination.

**Comment by outdoorbits** [comment]
It works now. Just update, setup cloud service GooglePhotos:

Basedir: upload
Method of data transfer: rclone
Storage location static: no

But remember, if you upload again it will upload all the photos again as they don't stay into the upload folder.
